hugolib/page.go

```diff
@@ -175,6 +175,19 @@ type Page struct {
 	// isn't accomanied by one.
 	sections []string
 
+	// Will only be set for section pages.
+	parent *Page
+
+	// Will only be set for regular pages.
+	currentSection *Page
```

Comment:
Remove `currentSection`.  `Page` simply needs a `parent *Page` regardless of Kind.